src/RecoveryServices/RecoveryServices.Backup/Cmdlets/Item/DisableAzureRmRecoveryServicesBackupProtection.cs

```diff
@@ -90,6 +98,11 @@ public override void ExecuteCmdlet()
                             throw new ArgumentException(String.Format(Resources.DisableWithRetainBackupNotCrititcal));
                         }
 
+                        if(DeleteBackupData && RetainRecoveryPointsAsPerPolicy.IsPresent)
+                        {
+                            throw new ArgumentException(String.Format(Resources.CantRemoveAndRetainRPsSimultaneously));
```

Comment:
Please use the ["AzPS" series of exceptions](https://github.com/Azure/azure-powershell-common/tree/main/src/Common/Exceptions) whenever possible. It helps with detailed telemetry data.